lib/kube/proxy/utils_testing.go

```diff
@@ -223,7 +223,12 @@ func SetupTestContext(ctx context.Context, t *testing.T, cfg TestConfig) *TestCo
 		// this is used to make sure that heartbeat started and the clusters
 		// are registered in the auth server
 		OnHeartbeat: func(err error) {
-			require.NoError(t, err)
+			// ignore not found errors because although the heartbeat is called before
+			// the close does not wait for the resource cleanup to finish.
+			if trace.IsNotFound(err) {
+				return
```

Comment:
It looks like there may be cases where this causes the test runner to hang, as when this heartbeat function is called, it does not trigger a test error and it does not put a value on the `heartbeatsWaitChannel` channel, so the code below that is waiting for all the heartbeats will miss one and block forever. This is assuming that there is only one heartbeat per cluster - which seems a little odd, as I would not expect this to be 1:1 - but I haven't seen this part of the code before now.

Reply:
You are correct but I don't think that would be a big issue since you cannot start the test without valid clusters 
Fixed it anw in [5986926](https://github.com/gravitational/teleport/pull/25441/commits/5986926072cacb483caec9120a0c32303a878525)

Reply:
I see now I misunderstood the structure of the test. I didn't realise this was all in the startup phase so the waiting was waiting for a single heartbeat from each cluster before continuing with the tests. It is only after that the error you are handling in this PR is generated - I'm happy for you to drop that extra commit if you like - I'm not sure it is needed and now makes the test setup a little more complicated than needed. Up to you.

Reply:
I will leave the last commit as it is since the check is strong. Thanks!